src/daemon/inout_route.rs

```diff
@@ -103,3 +101,66 @@ pub async fn out_route_obfsudp(
         .await;
     }
 }
+
+async fn per_route_tasks(
+    ctx: DaemonContext,
+    pipe: impl Pipe,
+    their_fp: Option<Fingerprint>,
+    link_price: LinkPrice,
+) -> anyhow::Result<()> {
+    let link_info =
+        LinkConnection::connect(ctx.clone(), pipe, link_price.max_outgoing_price).await?;
+
+    if let Some(fp) = their_fp {
+        let remote_fp = link_info.conn.remote_idpk().fingerprint();
+        log::info!("about to insert into neightable for fp: {}", fp);
+
+        if fp != remote_fp {
+            anyhow::bail!(
+                "out route fingerprint in config ({}), does not match link fingerprint: {}",
+                fp,
+                remote_fp,
+            );
+        }
+
+        link_info
+            .client
+            .push_price(link_price.incoming_price, link_price.incoming_debt_limit)
+            .await?;
+        ctx.get(DEBTS).insert_incoming_price(
+            link_info.conn.remote_idpk().fingerprint(),
+            link_price.incoming_price,
+            link_price.incoming_debt_limit,
+        );
+
+        ctx.get(NEIGH_TABLE)
+            .insert_pinned(fp, link_info.conn.clone());
```

Comment:
if we remove LinkConnection, then we'd be inserting a `Sender` here, since the NeighTable will simply map to outgoing channels.